build.sbt

```diff
@@ -39,6 +39,9 @@ lazy val root = (project in file("."))
     // Modern way to set Scala options
     Compile / scalacOptions ++= Seq("-deprecation", "-feature"),
 
+    // Scalafmt
+    scalafmtOnCompile := true,
```

Comment:
I think we should remove this. 
It will format every PR and contributes have to sync the PR to local every time.

Reply:
In spark it only check but write a error msg https://github.com/apache/spark/blob/dd153307cb9735fd05a41124eca2a136f40f3b3f/dev/lint-scala#L41

Reply:
What do you think @SemyonSinchenko ?

Reply:
I removed it.

Reply:
but how are we going to know if the code is in style then?

Reply:
> but how are we going to know if the code is in style then?

We should run `root/scalafmtCheckAll` as part of the CI. I can do it as a part of the current PR or I can open a new one. The idea of calling `scalafmtOnCompile := true` was exactly to avoid changes in CI and forcing `scalafmt` on running test or assembly.

Reply:
ok.